packages/pipeline-editor/src/utils.ts

```diff
@@ -59,16 +59,24 @@ export default class Utils {
   }
 
   /**
-   * Check if the provided pipeline is a newly created pipeline
+   * Check if the provided pipeline is a newly created pipeline (no nodes available)
    *
    * @param pipelineDefinition
    */
   static isNewPipeline(pipelineDefinition: any): boolean {
-    if (Object.keys(pipelineDefinition.pipelines[0].nodes).length == 0) {
-      return true;
-    } else {
-      return false;
-    }
+    return Object.keys(pipelineDefinition.pipelines[0].nodes).length === 0;
+  }
+
+  /**
+   * Check if the provided pipeline is clear of nodes and comments
+   *
+   * @param pipelineDefinition
+   */
+  static isEmptyCanvas(pipelineDefinition: any): boolean {
```

Comment:
There might be cases we want to check if canvas is completely clean of nodes and comments in order to enable/disable action buttons (eg. we only disable clean pipeline button if canvas is empty and there is nothing else left to clean)